File: medium/longest_substring_no_repeating.py
# Longest Substring w/o Repeating Characters

def longest(s):
    current = ""
    longest = 0

    for i in range(len(s)):
        if s[i] in current:
            start = current.index(s[i])
            current = current[start] + s[i]
        else:
            current += s[i]
        longest = max(len(current), longest)
    return longest

File: medium/longest_substring_no_repeating.js
// Longest Substring w/o Repeating Characters

function longest(s){
    let current = "";
    let [longest, start] = [0,0];
    for (let i = 0; i < s.length; i++) {
        start = current.indexOf(s[i]);
        if (start !== -1) {
            current = currentlslice(start + 1) + s[i]
        } else {
            current += s[i];
        }
        longest = Math.max(longest, current.length);
    }
    return longest;
}